cli/cmd/search.go

```diff
@@ -35,45 +81,33 @@ func NewSearchCmd() *cobra.Command {
 # Remote search across all 2.11 debug versions of Tarantool Enterprise Edition.
 
     $ tt search tarantool-ee --debug --version 2.11`,
-		Run: func(cmd *cobra.Command, args []string) {
-			err := modules.RunCmd(&cmdCtx, cmd.CommandPath(), &modulesInfo,
-				internalSearchModule, args)
-			handleCmdErr(cmd, err)
-		},
-		ValidArgs: []string{"tt", "tarantool", "tarantool-ee"},
 	}
 	searchCmd.Flags().BoolVarP(&local, "local-repo", "", false,
 		"search in local files")
-	searchCmd.Flags().BoolVar(&debug, "debug", debug,
-		"search for debug builds of tarantool-ee SDK")
-	searchCmd.Flags().StringVar(&searchCtx.ReleaseVersion, "version", searchCtx.ReleaseVersion,
-		"specify version")
+
+	searchCmd.AddCommand(
+		newSearchTarantoolCmd(),
+		newSearchTarantoolEeCmd(),
```

Comment:
So, we should remove manual options check in search implementation. For example, there is no need to check that dbg search is applicable only for ee. This is handled by sub-commands now.

Reply:
Removed.